### PR TITLE
JBPM-5980 & JBPM-5981: Bug fixings for shapes docking & delete node commands

### PR DESCRIPTION
**Pull request by romartin**
Hey @manstis @hasys ,

This commit provides fixes for these two JIRA tickets: [JBPM-5980](https://issues.jboss.org/browse/JBPM-5980) and [JBPM-5981](https://issues.jboss.org/browse/JBPM-5981).

It contains:
- Bug fixing on the canvas draw, and add child/node commands (JBPM-5980)
- Bug fixings on the delete node commands for both canvas and graph structures. (JBPM-5981)
- Added an improvement in the delete node that it shortcuts the incoming/outgoing connectors for a node, if applies, instead of keeping the connectors present but not connected.
- @manstis Refactored the containment/docking acceptor implementations for both default and CM implementations. There was some bug, no tests and the code was old and was not clear, so just doing clean implementations with new tests.
- Finally found a bug while testing the app that does not appear on the tickets, but using the keyboard to undo and redo actions, was causing issues when several editors were open.

Thanks guys!

### CODE REVIEW

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
hey @hasys  , I have verified the build in my local using either the `full` profile and I'm NOT able to reproduce the Jenkins's build error for this PR. Any ideas?? 
PS - Added a new commit due to just rebased from master `HEAD` again.
Thanks!

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by hasys** [comment]
Hi @romartin, looks like unrelated to PR fail. Will check again.

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
hey @hasys @manstis , there is not way this gets on blue.. :( you know if the compilation failure is also failing on other places?  I cannot reproduce it locally, no way. What can I do to have this merged? any clues? Thanks!

**Comment by manstis** [comment]
@romartin Hello.. there was a problem in ```droolsjbpm-integration``` that led to all downstream repositories' PRs to continually fail. @mariofusco says this is now fixed and PRs will begin again to pass. I assume however once a PR is tagged with "build" and "full downstream build" you may need both to pass to turn green (or blue.... are you colour blind?)

**Comment by romartin** [comment]
hey @manstis , hehe thanks, good point about the concrete color ... was just writing fast and... hehe :) Great thanks!

**Comment by romartin** [comment]
Jenkins please retest this.

**Comment by romartin** [comment]
hey @manstis @hasys , I have rebased again this branch from master `HEAD`, I can compile it locally even with `full` profile, so let's see if it gets green soon... (yes @manstis , green... :P )
Thanks!

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/session/impl/ClientFullSessionTest.java [hunk 1]
```diff
@@ -210,4 +261,13 @@ public void testDestroySession() {
         verify(builderControl,
                times(1)).disable();
     }
+
+    private void buildTestedInstance() {
+        ;
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/test/java/org/kie/workbench/common/stunner/core/client/session/impl/ClientFullSessionTest.java]: Floating semicolon

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/canvas/view/CanvasView.java [hunk 2]
```diff
@@ -143,39 +142,29 @@ public void init() {
...
-                             child);
-        }
+        dockShape(parentShape,
+                  childShape);
+        return this;
+    }
+
+    AbstractCanvas.View dockShape(final WiresContainer parentShape,
+                                  final WiresShape childShape) {
         parentShape.add(childShape);
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/canvas/view/CanvasView.java]: Not tested changes.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/ConnectionAcceptorControlImpl.java [hunk 3]
```diff
@@ -43,14 +42,12 @@
 import org.kie.workbench.common.stunner.core.graph.content.view.View;
 
 @Dependent
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/ConnectionAcceptorControlImpl.java]: It looks like a not tested class with bunch of changes.

**Comment by romartin** [comment]
Hey @hasys , yeah thanks for the feedback, agree about the missing tests as well, so just added those! Thanks again!

**Comment by romartin** [comment]
Hey @manstis @hasys , the build fails but I think it's not related to this commit,  it seems something missing on the kie-server side, see [my comments on this PR](https://github.com/kiegroup/droolsjbpm-integration/pull/966).
Anyway I had been able to test this by reverting the kie-server changes on my local.

**Comment by romartin** [comment]
Jenkins please retest this.

**Comment by romartin** [comment]
Hey @hasys , please can you give it a look again, and approve if you think it's good? I have another PR waiting for this..  lemme know if any issues as well please. thanks in advance!!

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
hey @manstis @hasys , I rebased this from master again - please this is starting to be urgent for me, could you please take it a look? The downstream build is continuously failing.. :(

**Comment by romartin** [comment]
Ohh green again finally! cc @manstis  @hasys  :)

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
hey @manstis  @hasys , please can we skip that Jenkins downstream build and merge this??

**Comment by romartin** [comment]
@manstis @hasys  this is blocking three other features to get in..... :(

**Comment by manstis** [comment]
@romartin I tend to agree with you; as ```jbpm-form-modeler``` follows ```kie-wb-common``` in a "full downstream build". That said it's plausible your PR breaks ```jbpm-form-modeler``` so I am quickly compiling ```jbpm-form-modeler``` (master) here w/o yuor PR to see if it fails too.. if it does then I will merge this. If it does not then I fear your PR has introduced the issue and it'll need fixing.

**Comment by manstis** [comment]
@pefernan Looks like ```jbpm-form-modeler``` is broken (on master). Fails to compile locally for me too.

@romartin You are in the clear!

**Comment by romartin** [comment]
@manstis oh great great, thanks Michael, really appreciated!! :)

**Pull request merged**